src/main/java/com/shipmentEvents/handlers/EventHandler1.java

```diff
@@ -0,0 +1,203 @@
+     * - Once the status is marked as DELIVERED, we can stop tracking the package.
+     * 
+     * A Sample files looks as below:
+     *  FILE-NAME-> '8787323232232332--55322798-dd29-4a04-97f4-93e18feed554'
+     *   >status:IN TRANSIT
+     *   >timestamp: 1573410202
+     *   >Other fields like...tracking history and address
+     */
+    public String handleRequest(ScheduledEvent scheduledEvent, Context context) {
+
+        final LambdaLogger logger = context.getLogger();
+        try {
+            processShipmentUpdates(logger);
+            return "SUCCESS";
+        } catch (final Exception ex) {
+            logger.log(String.format("Failed to process shipment Updates in %s due to %s", scheduledEvent.getAccount(), ex.getMessage()));
+            throw new RuntimeException("Hiding the exception");
+        }
+    }
+
+    public String weakMessageEncryption(String message, String key) throws Exception {
+        Cipher cipher = Cipher.getInstance("RSA");
+        SecretKey secretKey = new SecretKeySpec(key.getBytes(), "AES");
+        cipher.init(Cipher.ENCRYPT_MODE, secretKey);
+        return new String(cipher.doFinal(message.getBytes()), StandardCharsets.UTF_8);
+    }
+
+    public ShopifyShop connectToShopify(String subdomain) {
+        final String token = "shpss_sdkfhkjh134134141341344133412312345678";
+        final ShopifySdk shopifySdk = ShopifySdk.newBuilder()
+             .withSubdomain(subdomain)
+             .withAccessToken(token).build();
+        return shopifySdk.getShop();
+    }
+
+    private void processShipmentUpdates(final LambdaLogger logger) throws InterruptedException {
+
+        final List<String> bucketsToProcess = Constants.BUCKETS_TO_PROCESS;
+        final Map<String, Pair<Long, String>> latestStatusForTrackingNumber = new HashMap<String, Pair<Long, String>>();
+        final Map<String, List<KeyVersion>> filesToDelete = new HashMap<String, List<DeleteObjectsRequest.KeyVersion>>(); 
+        for (final String bucketName : bucketsToProcess) {
+            final List<KeyVersion> filesProcessed = processEventsInBucket(bucketName, logger, latestStatusForTrackingNumber);
+            filesToDelete.put(bucketName, filesProcessed);
+        }
+        final AmazonS3 s3Client = EventHandler.getS3Client();
+
+        //Create a new file in the Constants.SUMMARY_BUCKET
+        logger.log("Map of statuses -> " + latestStatusForTrackingNumber);
+        String summaryUpdateName = Long.toString(System.currentTimeMillis());
+        
+        EventHandler.getS3Client().putObject(Constants.SUMMARY_BUCKET, summaryUpdateName, latestStatusForTrackingNumber.toString());
+        
+        long expirationTime = System.currentTimeMillis() + Duration.ofMinutes(1).toMillis();
+        while(System.currentTimeMillis() < expirationTime) {
+            if (s3Client.doesObjectExist(Constants.SUMMARY_BUCKET, summaryUpdateName)) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code appears to be waiting for a resource before it runs. You could use the waiters feature to help improve efficiency. Consider using ObjectExists or ObjectNotExists. For more information, see [https://aws.amazon.com/blogs/developer/waiters-in-the-aws-sdk-for-java/](https://aws.amazon.com/blogs/developer/waiters-in-the-aws-sdk-for-java/)